### PR TITLE
Update TOC labels and hashIds

### PR DESCRIPTION
**Pull request by benhammondmusic**
## Preview Link

<a href="https://deploy-preview-1801--health-equity-tracker.netlify.app/exploredata?onboard=false" target="_blank">
  <br />
Click to demo updates </a> 

## Description

- switch to use "Population vs. Share" in the table of contents link to match the order the bars are displayed on the chart
- use "Unknown Demographic Map"
- use "&"
- update all hashIds to match #1761 

## Motivation and Context

We have taken care elsewhere with the order of key terms, as the chart can be confusing and it's important to display things in a consistent order to assist the user in understanding what they are looking at

## How has this been tested?

manually 

## Screenshots (if appropriate):
Screen Shot 2022-09-19 at 1 04 42 PM

## Types of changes

- Bug fix (non-breaking change which fixes an issue)

### CODE REVIEW

**Review by eriwarr — APPROVED** [review]
🚀

File: frontend/src/reports/ReportProviderSteps.ts [hunk 1]
```diff
@@ -2,43 +2,43 @@ import { StepData } from "../utils/hooks/useStepObserver";
...
+    hashId: "population-vs-share",
   },
   {
     label: "Data Table",
     pluralOnCompare: true,
-    hashId: "table",
+    hashId: "data-table",
   },
   {
     label: "Risk Ratio Table",
```
Comment (joshzarrabi) [inline frontend/src/reports/ReportProviderSteps.ts:35]: Can we rename this "Age-Adjusted Risk Ratio"

**Review by joshzarrabi — APPROVED** [review]
Nice

File: frontend/src/reports/ReportProviderSteps.ts (hunk 1 above)
Comment (benhammondmusic) [reply frontend/src/reports/ReportProviderSteps.ts:35]: good catch

**Pull request merged**